components/salsa-2022/src/function/accumulated.rs

```diff
@@ -65,23 +64,17 @@ impl Stack {
     /// Extend the stack of queries with the dependencies from `origin`.
     fn extend(&mut self, origin: Option<QueryOrigin>) {
         match origin {
-            None
-            | Some(QueryOrigin::Assigned(_))
-            | Some(QueryOrigin::BaseInput)
-            | Some(QueryOrigin::Field) => {}
+            None | Some(QueryOrigin::Assigned(_)) | Some(QueryOrigin::BaseInput) => {}
             Some(QueryOrigin::Derived(edges)) | Some(QueryOrigin::DerivedUntracked(edges)) => {
-                for DependencyIndex {
-                    ingredient_index,
-                    key_index,
-                } in edges.inputs().iter().copied()
+                for (_, dependency_index) in edges
+                    .input_outputs
```

Comment:
I removed the `inputs` method from `QueryEdges` and iterated directly over the field `input_ouputs`. Previously we needed to iterate over `input_outputs` to get `inputs` and then over `inputs`, plus it would create a `Vec` as a container for `inputs`. I think the current implementation would be a bit more efficient

Reply:
yes, but I wonder if we should add a method like

```rust
impl QueryEdges {
    pub fn inputs(&self) -> impl Iterator<Item = DependencyIndex> {
        self.iter().filter().copied()
    }
}
```

instead?